src/mame/mattel/pixtermu.cpp

```diff
@@ -0,0 +1,205 @@
+	if (data == 0) {
+		m_remap_view.select((m_apb[APB_PBC] & 0b0100) && (m_apb[APB_CS1OV] & 0b1) ? 0 : 4);
+	} else if (data < 4) {
+		m_remap_view.select(data);
+	} else {
+		logerror("Unexpected remap %d\n", data);
+	}
+}
+
+void pixter_multimedia_state::machine_start()
+{
+	if (m_cart->exists()) {
+		std::string region_tag;
+		memory_region *cart_rom = m_cart->memregion("cart:rom");
+		uint32_t const size = m_cart->common_get_size("rom");
+		m_maincpu->space(AS_PROGRAM).install_rom(0x48000000, 0x48000000 + size - 1, size, cart_rom->base());
+	}
+}
+
+void pixter_multimedia_state::machine_reset()
+{
+	m_apb[APB_PBC] = 0b0000; // Boot from NOR Flash or SRAM, 16-bit data bus width, nBLEx LOW for reads
+	m_apb[APB_CS1OV] = 0b0; // nCS1 is routed for normal operation
+	m_apb[APB_EPM] = 0b1111; // All external devices are accessible following reset
+
+	m_apb[APB_REMAP] = 0b00; // Map nCS1
+	apb_remap(m_apb[APB_REMAP]);
+}
+
+void pixter_multimedia_state::arm7_map(address_map &map)
+{
+	// Remap Bank
+	map(0x00000000, 0x003fffff).view(m_remap_view);
+	m_remap_view[0](0x00000000, 0x00001fff).rom().region("bootrom", 0);
+	m_remap_view[1](0x00000000, 0x003fffff).ram().share("ndcs0");
+	m_remap_view[2](0x00000000, 0x00003fff).ram().share("internal_sram");
+	m_remap_view[3](0x00000000, 0x003fffff).share("ncs0");
+	m_remap_view[4](0x00000000, 0x003fffff).rom().region("ncs1", 0);
+
+	// External SRAM
+	map(0x20000000, 0x203fffff).ram().share("ndcs0");
+	// nCS0 (Unused NAND Flash?)
+	map(0x40000000, 0x403fffff).share("ncs0");
+	// nCS1 (Chip-On-Board ROM)
+	map(0x44000000, 0x443fffff).rom().region("ncs1", 0);
+	// nCS2 (Cart ROM)
+	// map(0x48000000, 0x483fffff)
+	// nCS3 (Unused?)
+	// map(0x4c000000, 0x4c3fffff)
+
+	// Internal SRAM
+	map(0x60000000, 0x60003fff).mirror(0x0fffc000).ram().share("internal_sram");
+
+	// Boot ROM
+	map(0x80000000, 0x80001fff).rom().region("bootrom", 0);
+
+	// APB Bridge
+	map(0xfffc0000, 0xfffe6fff).ram().share("apb").w(FUNC(pixter_multimedia_state::apb_bridge_w));
+	// External Memory Control
+	map(0xffff1000, 0xffff1fff).ram();
+	// Color LCD Control
+	map(0xffff4000, 0xffff4fff).ram();
+	// USB Device
+	map(0xffff5000, 0xffff5fff).ram();
+	// Interrupt Vector Control
+	map(0xfffff000, 0xffffffff).ram();
+}
+
+void pixter_multimedia_state::apb_bridge_w(offs_t offset, uint32_t data, uint32_t mem_mask)
+{
+	if (offset == m_apb[APB_REMAP]) {
+		apb_remap(data);
+	}
+
+	COMBINE_DATA(&m_apb[offset]);
+}
+
+static INPUT_PORTS_START( pixter_multimedia )
+INPUT_PORTS_END
+
+void pixter_multimedia_state::pixter_multimedia(machine_config &config)
+{
+	// User's Guide - 1.3 Clock Strategy - AHB Fast CPU Clock (FCLK)
+	ARM7(config, m_maincpu, 76'205'000);
+	m_maincpu->set_addrmap(AS_PROGRAM, &pixter_multimedia_state::arm7_map);
+
+	GENERIC_CARTSLOT(config, m_cart, generic_plain_slot, "pixter_cart");
+	m_cart->set_endian(ENDIANNESS_LITTLE);
+	m_cart->set_width(GENERIC_ROM32_WIDTH);
+	m_cart->set_must_be_loaded(false);
```

Comment:
You’re still going to need `m_cart->set_device_load` to deal with loose software (i.e. not loaded from the software list).  It’s possible to just refuse to load it (in which case you need to actually check for that and return an error).  It’s better to support it though, as it makes it easier to test new dumps, develop homebrew software, etc. without needing to edit the software list for every change.

Reply:
Maybe I misunderstood you previous comment. I've added a guard if it was loaded via software list, to just return without doing any allocation, was that the intention?